street_network/src/transform/classify_intersections.rs

```diff
@@ -0,0 +1,46 @@
+use crate::osm::NodeID;
+use crate::IntersectionComplexity::*;
+use crate::{IntersectionComplexity, OriginalRoad, RawIntersection, StreetNetwork};
+
+/// Determines the initial complexity of all intersections. Intersections marked "Crossing" are
+/// considered "unclassified" and will be updated with a guess, others will be left unchanged.
+pub fn classify_intersections(streets: &mut StreetNetwork) {
+    let mut changes: Vec<(NodeID, IntersectionComplexity)> = Vec::new();
+    for (id, inter) in streets.intersections.iter() {
```

Comment:
Style nit
```suggestion
    for (id, inter) in &streets.intersections {
```

The separate loops to guess and then assign are unfortunately necessary due to `streets.roads_per_intersection`, as I'm sure the borrow checker let you notice. ;)  Even more motivation for me to finish refactoring and storing the set of roads on `RawIntersection` directly.

Reply:
It certainly did!

I did a lot of back and forth writing my street network impl on where to store the connections between roads and intersections in rust. Making references work out seems not worth it (due to lifetimes and the doubly linked nature of roads and intersections). The other option was using ids and getting roads and intersections out of a map every time, which actually meshes pretty well with some "data oriented" ideas that I'm vaguely familiar with, especially if the ids could be an index.

Working with ids, my favourite solution was to use a proper graph impl to store the source of truth for connectivity. It is the right abstraction and gives us all the `roads_per_intersection` type functionality that we could need. Its a bit wrinkly, but an undirected multigraph with edges around a node ordered in CW order, should cover us i think.

I'm interested to see where we go with this.